### PR TITLE
Stricter access control

### PR DESCRIPTION
**Pull request by danielbreves**
:koala:

/cc @zendesk/quokka

### Risks
 - None

### CODE REVIEW

File: lib/zendesk_apps_tools/server.rb [hunk 1]
```diff
@@ -6,9 +6,10 @@ class Server < Sinatra::Base
     set :protection, :except => :frame_options
     set :public_folder, proc { "#{settings.root}/assets" }
     last_mtime = Time.new(0)
+    ZENDESK_DOMAINS_REGEX = /^http(?:s)?:\/\/[a-z0-9-]+\.(?:zendesk|zopim|zd-(?:dev|master|staging))\.com$/
 
     get '/app.js' do
-      headers 'Access-Control-Allow-Origin' => '*'
+      headers 'Access-Control-Allow-Origin' => '*' if valid_origin
```
Comment (adammw) [inline lib/zendesk_apps_tools/server.rb]: `=request.env['HTTP_ORIGIN']`?

**Comment by danielbreves** [comment]
@zendesk/quokka :haircut:

**Comment by timbertson** [comment]
:+1:

**Comment by sandlerr** [comment]
setting it to the origin (instead of *) will send the session over, do you care? Shouldn't matter since its the local machine but also should be useless since it is encrypted.

**Comment by danielbreves** [comment]
@adammw any security concerns with leaving it '*'?

**Comment by adammw** [comment]
:+1:

**Comment by timbertson** [comment]
@danielbreves only thing I can think of is that the browser _might_ cache the "*" value and use it to allow future requests when served from cache. But mostly it just seems a bit improper.

**Pull request merged**